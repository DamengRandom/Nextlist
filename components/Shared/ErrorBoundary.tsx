import React from "react";

interface ErrorBoundaryProps {
  children: React.ReactNode;
}

interface ErrorBoundaryState {
  hasError: boolean;
  error?: Error;
}

class ErrorBoundary extends React.Component<ErrorBoundaryProps, ErrorBoundaryState> {
  constructor(props: ErrorBoundaryProps) {
    super(props);
    this.state = { hasError: false };
  }

  static getDerivedStateFromError(error: Error) {
    return { hasError: true, error };
  }

  componentDidCatch(error: Error, errorInfo: React.ErrorInfo) {
    console.error("ErrorBoundary caught an error", error, errorInfo);
  }

  render() {
    if (this.state.hasError) {
      return <div style={{ padding: 32, textAlign: 'center' }}><h2>Something went wrong.</h2><pre>{this.state.error?.message}</pre></div>;
    }
    return this.props.children;
  }
}

export default ErrorBoundary;
