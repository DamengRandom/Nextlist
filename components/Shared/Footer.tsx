import { Box, Text, Flex } from "@chakra-ui/react";

const Footer = () => {
  const currentYear = new Date().getFullYear();
  const version = "v1.0.1";

  return (
    <Box as="footer" bgGradient="linear(to-r, teal.500, blue.500)" color="white" boxShadow="md" p={4}>
      <Flex
        maxW="container.xl"
        mx="auto"
        alignItems="center"
        justifyContent="center"
      >
        <Text fontSize="sm">
          &copy; {currentYear} DamengRandom made with{" "}
          <Text as="span" color="red.500" mx={1}>&hearts;</Text> - {version}
        </Text>
      </Flex>
    </Box>
  );
};

export default Footer;
