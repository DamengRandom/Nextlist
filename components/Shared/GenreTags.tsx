import React from "react";
import { HStack, Box } from "@chakra-ui/react";

interface GenreTagsProps {
  genres: string[];
  spacing?: number;
  mb?: number;
}

const colors = [
  "teal.200",
  "blue.200",
  "pink.200",
  "orange.200",
  "purple.200",
  "green.200",
  "yellow.200"
];

export const GenreTags: React.FC<GenreTagsProps> = ({ genres, spacing = 1, mb = 2 }) => (
  <HStack spacing={spacing} mb={mb} flexWrap="wrap" data-testid="genre-tags-container">
    {genres?.map((genre, idx) => (
      <Box
        key={genre}
        px={2}
        py={0.5}
        bg={colors[idx % colors.length]}
        borderRadius="md"
        fontSize="xs"
        opacity={0.8}
        data-testid="genre-tag"
      >
        {genre}
      </Box>
    ))}
  </HStack>
);