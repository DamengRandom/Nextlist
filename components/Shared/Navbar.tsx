import { Flex, Box, Button, Heading, Spacer } from "@chakra-ui/react";
import { useAnimeStore } from "../../lib/store/animeStore";
import { useNavigation } from "../../hooks/useNavigation";

type NavbarProps = {
  handleEdit: () => void;
  username: string | null;
}

export default function Navbar({ handleEdit, username }: NavbarProps) {
  const { reset } = useAnimeStore();
  const { navigateTo } = useNavigation();

  const handleClick = () => {
    navigateTo('/');
    reset();
  }

  return (
    <Flex as="nav" align="center" justify="space-between" padding={4} bgGradient="linear(to-r, teal.500, blue.500)" color="white" boxShadow="md">
      <Box>
        <Heading size="lg" textShadow="0 1px 2px rgba(0,0,0,0.5)" cursor="pointer" onClick={handleClick}>Nextlist</Heading>
      </Box>
      <Spacer />
      <Box>
        <Button
          aria-pressed={!!username}
          onClick={handleEdit}
          variant="outline"
          color="white"
          border="2px solid transparent"
          backgroundClip="padding-box"
          position="relative"
          _before={{
            content: '""',
            position: "absolute",
            top: 0,
            right: 0,
            bottom: 0,
            left: 0,
            zIndex: -1,
            margin: "-2px",
            borderRadius: "inherit",
            background: "linear-gradient(45deg, var(--chakra-colors-pink-500), var(--chakra-colors-blue-400))",
            opacity: 0,
            transition: "opacity 0.3s",
          }}
          _hover={{
            _before: {
              opacity: 1,
            },
            transform: "scale(1.02)",
          }}
        >
          {username ? `Hello ${username}` : "Sign In"}
        </Button>
      </Box>
    </Flex>
  );
}