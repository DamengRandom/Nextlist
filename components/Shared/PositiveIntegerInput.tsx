import { Input, InputProps } from "@chakra-ui/react";
import { ChangeEvent } from "react";

interface PositiveIntegerInputProps extends Omit<InputProps, 'onChange'> {
  value: string;
  onChange: (value: string) => void;
  maxLength?: number;
  width?: string | Record<string, string>;
}

const PositiveIntegerInput = ({
  value,
  onChange,
  maxLength = 4,
  width = "100px",
  ...props
}: PositiveIntegerInputProps) => {
  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
    const inputValue = e.target.value;

    if (inputValue === "") {
      onChange("");
      return;
    }
    
    if ((/^[1-9]\d*$/.test(inputValue) || /^0$/.test(inputValue)) && inputValue.length <= maxLength) {
      onChange(inputValue);
    }
  };

  return (
    <Input
      type="text"
      inputMode="numeric"
      pattern="[0-9]*"
      value={value}
      onChange={handleChange}
      size="sm"
      width={width}
      maxLength={maxLength}
      {...props}
    />
  );
};

export default PositiveIntegerInput;
