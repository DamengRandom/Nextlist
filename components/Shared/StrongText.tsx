import { Text, TextProps } from '@chakra-ui/react';

const StrongText = ({ children, ...props }: TextProps) => (
  <Text as="span" fontWeight="bold" display="inline" {...props}>
    {children}
  </Text>
);

export default StrongText;
