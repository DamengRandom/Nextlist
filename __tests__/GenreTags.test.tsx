import { render, screen } from '@testing-library/react';
import { GenreTags } from '../components/Shared/GenreTags';
import { ChakraProvider } from '@chakra-ui/react';

describe('GenreTags', () => {
  const testGenres = ['Action', 'Comedy', 'Drama', 'Sci-Fi'];
  const testSingleGenre = ['Thriller'];

  const renderWithChakra = (ui: React.ReactElement) => {
    return render(<ChakraProvider>{ui}</ChakraProvider>);
  };

  it('renders all genre tags', () => {
    renderWithChakra(<GenreTags genres={testGenres} />);
    testGenres.forEach(genre => {
      expect(screen.getByText(genre)).toBeInTheDocument();
    });
  });

  it('renders nothing when genres array is empty', () => {
    const { container } = renderWithChakra(<GenreTags genres={[]} />);
    expect(container.firstChild).toBeEmptyDOMElement();
  });

  it('handles undefined genres prop gracefully', () => {
    const { container } = renderWithChakra(<GenreTags genres={undefined as never} />);
    expect(container.firstChild).toBeEmptyDOMElement();
  });

  it('wraps tags when they overflow', () => {
    renderWithChakra(<GenreTags genres={testGenres} />);
    const hstack = screen.getByTestId('genre-tags-container');
    expect(hstack).toHaveStyle('flex-wrap: wrap');
  });

  it('matches snapshot with multiple genres', () => {
    const { asFragment } = renderWithChakra(<GenreTags genres={testGenres} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot with single genre', () => {
    const { asFragment } = renderWithChakra(<GenreTags genres={testSingleGenre} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot with empty genres', () => {
    const { asFragment } = renderWithChakra(<GenreTags genres={[]} />);
    expect(asFragment()).toMatchSnapshot();
  });
});