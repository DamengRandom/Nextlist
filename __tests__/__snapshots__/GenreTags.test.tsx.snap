// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreTags matches snapshot with empty genres 1`] = `
<DocumentFragment>
  <div
    class="chakra-stack css-1dth9i4"
    data-testid="genre-tags-container"
  />
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`GenreTags matches snapshot with multiple genres 1`] = `
<DocumentFragment>
  <div
    class="chakra-stack css-1dth9i4"
    data-testid="genre-tags-container"
  >
    <div
      class="css-ffmuqs"
      data-testid="genre-tag"
    >
      Action
    </div>
    <div
      class="css-1loqlts"
      data-testid="genre-tag"
    >
      Comedy
    </div>
    <div
      class="css-xzkf3k"
      data-testid="genre-tag"
    >
      Drama
    </div>
    <div
      class="css-1sl5dkm"
      data-testid="genre-tag"
    >
      Sci-Fi
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`GenreTags matches snapshot with single genre 1`] = `
<DocumentFragment>
  <div
    class="chakra-stack css-1dth9i4"
    data-testid="genre-tags-container"
  >
    <div
      class="css-ffmuqs"
      data-testid="genre-tag"
    >
      Thriller
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
